src/wallet/wallet.h

```diff
@@ -1363,7 +1369,7 @@ class CWallet : public CCryptoKeyStore, public CValidationInterface
     //! spending key to the wallet if it has not alreay been persisted.
     //! Returns the newly created key, and a flag distinguishing
     //! whether or not the key was already known by the wallet.
```

Comment:
The motivation for the change here is that the SaplingExtendedSpendingKey provides too many capabilities to the caller - namely, it could be used to generate a change address that would never have its keys added to the wallet.